common/djangoapps/course_modes/views.py

```diff
@@ -82,6 +82,16 @@ def get(self, request, course_id, error=None):
                 )
             )
 
+        # Check if the no-id-professional mode is added in course. If so then redirect to payment/verification flow
+        # skipping the verification step
+        no_id_prof_mode = CourseMode.mode_for_course(course_key, CourseMode.NO_ID_PROFESSIONAL_MODE[0])
+        if no_id_prof_mode:
+            return redirect(
+                reverse(
+                    'verify_student_payment_confirmation',
```

Comment:
I think this should still go to `verify_student_start_flow` -- the payment confirmation page is for *after* a user has paid.